manager/src/main/java/com/redhat/service/smartevents/manager/BridgesServiceImpl.java

```diff
@@ -90,6 +92,61 @@ public Bridge createBridge(String customerId, String organisationId, String owne
         return bridge;
     }
 
+    @Override
+    @Transactional
+    public Bridge updateBridge(String bridgeId, String customerId, BridgeRequest bridgeRequest) {
+        // Extract existing definition
+        Bridge existingBridge = getBridge(bridgeId, customerId);
+
+        if (!existingBridge.isActionable()) {
+            throw new BridgeLifecycleException(String.format("Bridge with id '%s' for customer '%s' is not in an actionable state.",
+                    bridgeId,
+                    customerId));
+        }
+        // We don't want to allow updating an Error Handler if the previous configuration has not been provisioned.
+        // This can lead to problems trying to remove the Error Handler if it is not in READY state.
+        ListResult<Processor> hiddenProcessors = processorService.getHiddenProcessors(bridgeId, customerId);
+        if (!hiddenProcessors.getItems().stream().allMatch(ManagedResource::isActionable)) {
```

Comment:
This is interesting....

To make things "simple" we cannot allow an Error Handler to be updated unless it is `READY`.

However Users know nothing of Error Handler processor provisioning as we chose to keep it internal.

Any better ideas to how to solve the problem?

We could make the `Bridge` stay in `PREPARING` until it's Error Handler is `READY` instead.

Reply:
Fine for me to keep `Bridge` in `PROVISIONING` state until all the components are fully provisioned.

Reply:
We had an initial discussion about this with @kelvah because I agree that we need to make this time to apply a change visible to users.
I agree that we can leverage `Bridge` intermediate statuses to represent this.

Reply:
Just to be clear on the implications for the Bridge `status` lifecycle here:

- If the error handler fails to deploy, is the containing Bridge instance marked as `FAILED`? If not, how does the user know if their error handler is broken if we failed to deploy it?
- If a user updates an error handler, and it fails to deploy, is the containing Bridge instance marked as `FAILED`?

In particular, that last scenario doesn't feel correct to me. I don't think we'd want to mark a bridge as `FAILED` if the user does a bad update to the error handler.

It feels like the error handler needs to be its own resource, perhaps at `bridges/{id}/error_handler` so that we can separate the lifecycle of the error_handler from the bridge instance on which is it deployed. That would mean that the user can determine if the error handler is working (or not) and update it without impacting the status of the Bridge e.g making a bad update to the error handler doesn't cause a bridge to go into a `FAILED` status.

Reply:
Hi @robpblake well...... 

The original Error Handler proposal had the `Processors` listed a long with every other `SINK` or `SOURCE` `Processor` however "we" deemed it unsuitable for Users to be aware of internal implementation details and hence hid Error Handler `Processor`'s from view... 

That was OK when you could only create them.

Now we need to support updating the Error Handler definition (or even removing or adding it entirely) we have to prevent the operation if the underlying `Processor` is not `READY` otherwise we run into all sorts of race condition issues. 

This PR, whilst supporting updating the Error Handler, does change the life-cycle to exactly as you fear.

An Error Handler `Processor` is now "just" a dependency of the `Bridge`... hence, as with other dependencies (i.e. RHOC and `Processor`'s), if provisioning of the dependency fails the parent is moved into a `FAILED` state too. Meaning both of your horror scenarios are true.

It sounds like we need a meeting (you, @danielezonca, @kelvah, me and anyone else interest....)